shopinvader_search_engine/models/se_indexable_record.py

```diff
@@ -16,3 +16,25 @@ def _filter_by_index(self):
                 in rec.se_binding_ids.mapped("index_id").ids
             )
         return records
+
+    def _filter_by_bound_in_same_lang(self):
```

Comment:
Could you please explain why, while working on a particular index, you would need to return records linked to another index (with the same backend and lang) ?

Reply:
when you serialize a product, you also serialize categories linked to the product. In this case, you're working with the product's index  not the one from the category. If you add tests to check that the categories field is filled into the product information, you'll see that before this change it was always empty.

Reply:
Ok, but now if you have two indexes for products for the same lang, products of one index will appear in the other and vice versa.

But I guess it doesn't make sense to have two indexes for the same model for the same lang thus we can accept that.

Reply:
There was a test, but it was wrong. I've corrected it (+ added one for the lang case).

I have also opened https://github.com/OCA/search-engine/pull/177 to avoid such mistakes in the future.

Reply:
@lmignon ping